spec/unit/plugins/linux/network_spec.rb

```diff
@@ -1335,6 +1335,42 @@
         end
       end
 
+      describe "when there's a source field in a local route entry but no dev" do
+        let(:linux_ip_route) do
+          <<~EOM
+            default proto bird src 172.16.19.39 metric 10 \\      nexthop via 10.1.81.1 dev eth0 weight 20
+            default via 10.116.201.1 dev eth0 metric 10 src 10.116.201.1
+            10.116.201.0/24 dev eth0  proto kernel  src 10.116.201.76
+            192.168.0.0/24 dev eth0  proto kernel  src 192.168.0.2
+            default via 10.116.201.254 dev eth0 metric 20 src 10.116.201.74
```

Comment:
It's unclear to me why we pick the no-dev route here over the dev route at the same metric.

Sorry, we're heading into parts of routing that I'm not super familiar with. Do we always prefer a no-dev route over a dev route at the same metric? Or are we picking it some other way? Based on the code it seems like we just always prefer no-dev routes?

Reply:
woops, good catch. in this specific case the route on line 1342 is being discarded by the condition here https://github.com/chef/ohai/blob/main/lib/ohai/plugins/linux/network.rb#L135
however, your question is valid. I made it so we explicitly rank routes that define a device over ones that don't if they have the same metric. if theres a specific part you'd like more comments/explanation on let me know I can try to add/explain further